compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1210,6 +1206,12 @@ public void visit(BLangXMLNS xmlnsNode) {
         defineSymbol(xmlnsNode.prefix.pos, xmlnsSymbol);
     }
 
+    private void checkInvalidNameSpaceDeclaration(Location pos, BLangIdentifier prefix, String nsURI) {
+        if (!nullOrEmpty(prefix.value) && nsURI.isEmpty()) {
```

Comment:
Why do we need `nullOrEmpty(prefix.value)` check?

Reply:
This check is used to allow the following scenario,
```ballerina
xmlns "";
```
When xmlns is defined without a prefix and has an empty string for the namespace-uri. Based on the [xml spec](https://www.w3.org/TR/xml-names/#iri-use),
this seems to be an issue as,
```
The empty string, though it is a legal URI reference, cannot be used as a namespace name.
```

Reply:
Created [issue](https://github.com/ballerina-platform/ballerina-lang/pull/41958) to track this